integration/setup.go

```diff
@@ -181,12 +181,17 @@ func (ts *testSetup) writeBatch(dm dataMap) error {
 	return m3dbClientWriteBatch(ts.m3dbClient, ts.workerPool, dm)
 }
 
-func (ts *testSetup) fetch(req *rpc.FetchRequest) (*rpc.FetchResult_, error) {
+func (ts *testSetup) fetch(req *rpc.FetchRequest) ([]m3db.Datapoint, error) {
 	if ts.opts.GetUseTChannelClientForReading() {
-		return tchannelClientFetch(ts.tchannelClient, ts.opts.GetReadRequestTimeout(), req)
+		datapoints, err := tchannelClientFetch(ts.tchannelClient, ts.opts.GetReadRequestTimeout(), req)
+		if err != nil {
+			return nil, err
+		}
+		// Sorting is necessary in that fetching with tchannel client may return out-of-order datapoints.
+		sort.Sort(byTimestampAscending(datapoints))
```

Comment:
Sorting is not necessary now, we actually create a multi-iterator, etc in the `Fetch` service now so we can avoid this sort here.